File: src/forza/telemetry/ForzaTelemetryBuilder.java
package forza.telemetry;

public class ForzaTelemetryBuilder {
    ForzaInterface listener;
    Thread connectionThread;
    int port = 5300;

    public ForzaTelemetryBuilder(){} //Used for default
    public ForzaTelemetryBuilder(int port) {
        this.port = port;
    }

    public ForzaTelemetryBuilder addListener(ForzaInterface listener) {
        this.listener = listener;
        return this;
    }

    public Thread getThread() {
        if(connectionThread == null) {
            connectionThread = listener.startConnection(port);
        }
        return connectionThread;
    }
}

File: kotlin_lib/ForzaTelemetryBuilder.kt
package com.fasetto.fragmentjava.kotlin_lib

class ForzaTelemetryBuilder {
    var listener: ForzaInterface? = null
    var port = 5300

    constructor() {} //Used for default
    constructor(port: Int) {
        this.port = port
    }

    fun addListener(listener: ForzaInterface?): ForzaTelemetryBuilder {
        this.listener = listener
        return this
    }

    val thread: Thread?
        get() = listener!!.startConnection(port)
}